npm2deb/__init__.py

```diff
@@ -85,6 +87,90 @@ def start(self):
         self.create_itp_bug()
 
+    def initiate_build(self ,saved_path):
+        """
+        Try building deb package after creating required files using start().
+        'uscan', 'uupdate' and 'dpkg-buildpackage' are run if debian/watch is OK.
+        """
+        uscan_info = self.test_uscan()
+        if uscan_info[0] == 0:
+            self.run_uscan()
+
+            for name in _os.listdir('..'):
+                orig_file = _re.search('%s_(\d.*)\.orig\.(?:zip|tgz|tbz|txz|(?:tar\.(?:gz|bz2|xz)))' % self.debian_name, name)
+                if orig_file:
+                    orig_file = orig_file.group(0)
+                    break
+            self.run_uupdate(orig_file)
+
+            new_dir = orig_file[:orig_file.find('.orig')].replace('_','-')
+            utils.change_dir('../%s' % new_dir)
+            self.run_buildpackage()
+            self.edit_changelog()
+
+            debian_path = "%s/%s/debian" % (self.name, new_dir)
+            print ('\nRemember, your new source directory is %s/%s' % (self.name, new_dir))
+
+        else:
+            debian_path = "%s/%s/debian" % (self.name, self.debian_name)
+
+        print("""
+This is not a crystal ball, so please take a look at auto-generated files.\n
+You may want fix first these issues:\n""")
+
+        utils.change_dir(saved_path)
+        _call('/bin/grep --color=auto FIX_ME -r %s/*' % debian_path, shell=True)
+
+        if uscan_info[0] != 0:
+            print ("\nUse uscan to get orig source files. Fix debian/watch and then run\
+                    \n$ uscan --download-current-version\n")
+
+        if self.upstream_watch:
+            print ("""
+*** Warning ***\nUsing fakeupstream to download npm dist tarballs, because upstream
+git repo is missing tags. Its better to ask upstream to tag their releases
+instead of using npm dist tarballs as dist tarballs may contain pre built files
+and may not include tests.""")
+
+    def edit_changelog(self):
+        """
+        This function is to remove extra line '* New upstream release'
+        from debian/changelog
+        """
+        with open('debian/changelog', 'r') as f:
+            data = f.read()
+        f.close()
+
+        data_list = data.split('\n')
+        data_list.pop(3)
+
+        with open('debian/changelog', 'w') as f:
+            f.write('\n'.join(data_list))
+        f.close()
+
+    def run_buildpackage(self):
+        print ("\nBuilding the binary package")
+        _call('dpkg-source -b .', shell=True)
+        _call('dpkg-buildpackage', shell=True)
+        # removing auto generated temporary files
+        _call('fakeroot debian/rules clean', shell=True)
+
+    def run_uupdate(self, tar_file):
+        print ('\nCreating debian source package...')
+        _call('uupdate -b ../%s' % tar_file, shell=True)
+
+    def run_uscan(self):
+        print ('\nDownloading source tarball file using debian/watch file...')
+        _os.system('uscan --download-current-version')
```

Comment:
You can add --download-version <current version in changelog> to your uscan command to make sure latest release is tagged.